frugalos_segment/src/client/mds.rs

```diff
@@ -164,9 +171,9 @@ impl MdsClient {
     ) -> impl Future<Item = (ObjectVersion, bool), Error = Error> {
         debug!(self.logger, "Starts PUT: id={:?}", id);
         let put_content_timeout = Seconds(if let Deadline::Within(d) = deadline {
-            d.as_secs() + 60 // XXX: hard-coding
+            d.as_secs() + self.client_config.minimum_put_content_timeout.0
         } else {
-            60
+            self.client_config.default_put_content_timeout.0
```

Comment:
For `Deadline::Immediate`, this case should be treated as `d == 0` rather than "the default case". Thus `self.client_config.minimum_put_content_timeout.0` seems to be more appropriate than `self.client_config.default_put_content_timeout.0`.
(The original code did not treat `Deadline::Infinity` well, thought...)

Reply:
Ok,  I've got the message. I'll change the code to use `self.client_config.minimum_put_content_timeout` in both cases.